compiler/deSugar/DsExpr.hs

```diff
@@ -713,6 +713,7 @@ ds_expr _ expr@(RecordUpd { rupd_expr = record_expr, rupd_flds = fields
                                          , pat_arg_tys = in_inst_tys
                                          , pat_wrap = req_wrap }
            ; return (mkSimpleMatch RecUpd [pat] wrapped_rhs) }
+ds_expr _ RecordUpd {} = panic "DsExpr: ds_expr: The impossible happened!"
```

Comment:
I think it would be useful to copy the comment or link to it in the upstream PR https://github.com/ghc/ghc/commit/cf65cf16c89414273c4f6b2d090d4b2fffb90759#

Reply:
done